### ISSUE TITLE
Wrong examples in the readme.md?

### ISSUE THREAD

**Issue by cap5lut** [AlessandroMinoccheri/UserPermissions#5]
Hi, I was checking out your plugin today.

First of all: Good job, I like it!

I had some struggles figuring out what failed, while going the the examples written in the readme.

I'm working with CakePHP 3.5 and PHP 7.1:
I got always the error message that my view checking function names were called on a string, I marked the line that failed:
```
$rules = array(
	'user_type' => $user_type,
	'redirect' => '/projects/',
	'message' => 'You do not have permission to access this page',
	'action' =>  $this->request->params['action'],
	'controller' =>  $this->request->params['controller'], // HERE ################
	'groups' => array(
		'guest' => array('register', 'logout', 'login'),
		'admin' => array('*'), 
		'admin-team' => array('register', 'add', 'logout', 'index', 'edit'),
		'user' => array('register', 'add', 'logout', 'index')
	),
	'views' => array(
		'edit' => 'checkEdit',
		'delete' => 'checkDelete',
	),
);
```

`$rules["controller"]` actually had to contain the reference to the instance of the controller, but `$this->request->params['controller']` just contains its name. (Thats not clear in the example)

Regards, cap5lut

**Comment by AlessandroMinoccheri** [comment]
I'm trying to investigate on It, if you already know how to solve It open a Pull request.

I hope in a few days to fix this bug,

Thanks for your contribute

**Comment by AlessandroMinoccheri** [comment]
Can you please post me the full error message please @cap5lut ?

**Comment by cap5lut** [comment]
If you follow the example code, on this line the code will fail:
[if(!$this->controller->$value()){](https://github.com/AlessandroMinoccheri/UserPermissions/blob/bd0cf6a9413032ebba42b255a31fcb31fb9693f3/src/Controller/Component/UserPermissionsComponent.php#L154)

Because in that scope `$this->controller` will be the data passed to `$this->UserPermissions->allow($rules);` by `$rules["controller"]`, which is set in the example by `$this->request->params['controller']`, which will just return the name of the controller (so a string). To fix it you will have to set `$rules["controller"]` to the reference of the controller itself (for example, `$rules["controller"] = $this;`

Here your example fixed up:
```
<?php
namespace App\Controller;

use App\Controller\AppController;

class ExampleController extends AppController
{
	public $components = array(
		'UserPermissions.UserPermissions'
	);
	
	public function beforeFilter () {
		parent::beforeFilter(); 
		
		$user_type = $this->Auth->user("role"); // just an example to generate rules.
		
		$rules = array(
			'user_type' => $user_type,
			'redirect' => '/projects/',
			'message' => 'No permission',
			'action' =>  $this->request->params['action'],
			'controller' =>  $this, // was previously $this->request->params['controller'],
			'groups' => array(
				'guest' => array('register', 'logout', 'login'),
				'admin' => array('*'), 
				'admin-team' => array('register', 'add', 'logout', 'index', 'edit'),
				'user' => array('register', 'add', 'logout', 'index')
			),
			'views' => array(
				'edit' => 'checkEdit',
				'delete' => 'checkDelete',
			),
		);

		$this->UserPermissions->allow($rules);
	}
}
```

**Comment by AlessandroMinoccheri** [comment]
Ok I'll fix the documentation so 

Thanks

**Comment by AlessandroMinoccheri** [comment]
Fixed into 3.0.2 version
